src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -134,17 +142,53 @@ public override void Initialize (AnalysisContext context)
 					CheckCalledMember (operationContext, methodSymbol, dangerousPatterns);
 				}, OperationKind.DelegateCreation);
 
+				static void CheckStaticConstructors (OperationAnalysisContext operationContext,
+					ImmutableArray<IMethodSymbol> constructors)
+				{
+					foreach (var constructor in constructors) {
+						if (constructor.Parameters.Length == 0 && constructor.HasAttribute (RequiresAssemblyFilesAttribute) && constructor.MethodKind == MethodKind.StaticConstructor) {
+							if (constructor.TryGetAttributeWithMessageOnCtor (FullyQualifiedRequiresAssemblyFilesAttribute, out AttributeData? requiresUnreferencedCode)) {
+								operationContext.ReportDiagnostic (Diagnostic.Create (
+									s_requiresAssemblyFilesRule,
+									operationContext.Operation.Syntax.GetLocation (),
+									constructor.ToString (),
+									(string) requiresUnreferencedCode!.ConstructorArguments[0].Value!,
+									requiresUnreferencedCode!.NamedArguments.FirstOrDefault (na => na.Key == "Url").Value.Value?.ToString ()));
+							}
+						}
+					}
+				}
+
 				static void CheckCalledMember (
 					OperationAnalysisContext operationContext,
 					ISymbol member,
 					ImmutableArray<ISymbol> dangerousPatterns)
 				{
+					// Find containing symbol
+					ISymbol? containingSymbol = null;
```

Comment:
Isn't this `FindAttributableParent`?

Reply:
I don't think they are the same FindAttributableParent search for a Method/Property/Event/Field to put the attribute on, this function only looks for local functions and lambdas to warn and otherwise falls back to operationContext.ContainingSymbol

Reply:
Lambdas can't have attributes so I'm not clear on what this is supposed to be doing.